### ISSUE TITLE
The self.pruned_weight in /dcp/mask_conv.py is not loaded from the original module.

### ISSUE THREAD

**Issue by Sahala08** [SCUT-AILab/DCP#22]
Hello, what a good work you did! 
I have the problem that when I reproduce the work, I find that the self.pruned_weight (in line 25 of /dcp/mask_conv.py) is not loaded from the original module (It may be generated from lib of nn.Conv2d? because the MaskConv2d derives from nn.Conv2d). 
The weight of original layer: 
![image](https://github.com/SCUT-AILab/DCP/assets/54014960/64073e2e-8254-450f-a3ea-065201a700cc)
The weight of self.pruned_weight:
![image](https://github.com/SCUT-AILab/DCP/assets/54014960/4be40c4a-94db-4a8a-bd5a-66161dc6ab1e)

Why the weight is not loaded from the original module? 
Thanks a lot.

**Comment by liujingcs** [comment]
The absence of initial initialization for pruned_weight occurs when Conv2d is replaced with MaskConv2d. Initially, pruned_weight is set to zero, as can be seen in the channel_selection.py file at https://github.com/SCUT-AILab/DCP/blob/70a2e53ae896573b0b4323eac5817e5660315cb4/dcp/channel_selection/channel_selection.py#L120. Subsequently, it undergoes optimization through gradient descent.

**Comment by Sahala08** [comment]
> The absence of initial initialization for pruned_weight occurs when Conv2d is replaced with MaskConv2d. Initially, pruned_weight is set to zero, as can be seen in the channel_selection.py file at
> 
> https://github.com/SCUT-AILab/DCP/blob/70a2e53ae896573b0b4323eac5817e5660315cb4/dcp/channel_selection/channel_selection.py#L120
> 
> . Subsequently, it undergoes optimization through gradient descent.

Ok, my bad. :joy: Thanks!
